src/packages/database/relationship/utils/getters.js

```diff
@@ -1,28 +1,63 @@
 // @flow
+import { camelize } from 'inflection';
+
 import type { Model } from '../../index';
 import type { Relationship$opts } from '../index';
 
 /**
  * @private
  */
+async function getHasManyThrough(owner: Model, {
```

Comment:
Not properly fetching `hasManyThrough` relationships is what caused all records to be returned as opposed to the actual value for the second (lazy loaded) relationship query that occurred due to the bug in the line comment above.